components/calendar/src/julian.rs

```diff
@@ -74,17 +74,50 @@ impl Calendar for Julian {
     }
 
     fn offset_date(&self, date: &mut Self::DateInner, offset: DateDuration<Self>) {
-        // TODO
+        date.year.0 += offset.years;
+        date.add_months(offset.months);
+        offset.months = 0;
+
+        offset.days += offset.weeks * 7;
+
+        // Normalize date to beginning of month
+        offset.days += date.day.0 as i32 - 1;
+        date.day.0 = 1;
+
+        while offset.days != 0 {
+            if offset.days < 0 {
+                date.add_months(-1);
+                let month_days = self.days_in_month(date);
+                if (-offset.days) > month_days as i32 {
+                    offset.days += month_days as i32;
+                } else {
+                    // Add 1 since we are subtracting from the first day of the
+                    // *next* month
+                    date.day.0 = 1 + (month_days as i8 + offset.days as i8) as u8;
+                    offset.days = 0;
+                }
+            } else {
+                let month_days = self.days_in_month(date);
+                if offset.days > month_days as i32 {
+                    date.add_months(1);
+                    offset.days -= month_days as i32;
+                } else {
+                    date.day.0 += offset.days as u8;
+                    offset.days = 0;
+                }
+            }
+        }
     }
 
+    #[allow(clippy::field_reassign_with_default)]
     fn until(
         &self,
         date1: &Self::DateInner,
         date2: &Self::DateInner,
         largest_unit: DateDurationUnit,
         smallest_unit: DateDurationUnit,
     ) -> DateDuration<Self> {
-        // TODO
+        Iso.until(&date1.0, &date2.0, largest_unit, smallest_unit).cast_unit()
```

Comment:
This isn't correct since the difference between ISO Feb 1 and ISO March 1 on a year which is leap in Julian but not Gregorian (e.g. 1900) is 29 days in Julian and 28 in Gregorian

Reply:
I think the current impl would return the difference as 1 month for ISO, rather than 28/29 days 🤔  

On checking the iso, I only see that the year, month and day are subtracted and that value is returned in an object. For that type of implementation, I think it is kinda ok to reuse it.

But yeah, its kinda wrong impl, will do julians own impl

Reply:
Both the offset and until functions should be generically extracted so they can share an implementation

Reply:
Sure, so keeping this as it is then.